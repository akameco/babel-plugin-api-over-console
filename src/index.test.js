// @flow
import pluginTester from 'babel-plugin-tester'
import plugin from '.'

pluginTester({
  plugin,
  snapshot: true,
  tests: [
    `console.log('hello')`,
    `function hello () {}`,
    `
    var a = 1
    console.log(a)
  `,
    `
    const obj = {a: 1, b: 2, c: 3}
    console.log(obj)
  `,
    `
    console.log('hello')
    console.log('world')
  `,
    `
    const args = process.argv.slice(2)
    console.log({name: args[1]})
  `,
    `console.log()`,
  ],
})
