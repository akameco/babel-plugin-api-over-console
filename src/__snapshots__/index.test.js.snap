// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api-over-console 1. api-over-console: 1. api-over-console 1`] = `
"
console.log('hello')

      ↓ ↓ ↓ ↓ ↓ ↓

const express = require('express');

const cors = require('cors');

const app = express();
const port = process.env.PORT || 3000;
app.use(cors());
app.get('/', (req, res) => {
  res.json('hello');
});
app.listen(port, () => {
  console.log('listening on %s', port);
});
"
`;

exports[`api-over-console 2. api-over-console: 2. api-over-console 1`] = `
"
function hello () {}

      ↓ ↓ ↓ ↓ ↓ ↓

function hello() {}
"
`;

exports[`api-over-console 3. api-over-console: 3. api-over-console 1`] = `
"
var a = 1
console.log(a)

      ↓ ↓ ↓ ↓ ↓ ↓

const express = require('express');

const cors = require('cors');

const app = express();
const port = process.env.PORT || 3000;
app.use(cors());
app.get('/', (req, res) => {
  var a = 1;
  res.json(a);
});
app.listen(port, () => {
  console.log('listening on %s', port);
});
"
`;

exports[`api-over-console 4. api-over-console: 4. api-over-console 1`] = `
"
const obj = {a: 1, b: 2, c: 3}
console.log(obj)

      ↓ ↓ ↓ ↓ ↓ ↓

const express = require('express');

const cors = require('cors');

const app = express();
const port = process.env.PORT || 3000;
app.use(cors());
app.get('/', (req, res) => {
  const obj = { a: 1, b: 2, c: 3 };
  res.json(obj);
});
app.listen(port, () => {
  console.log('listening on %s', port);
});
"
`;

exports[`api-over-console 5. api-over-console: 5. api-over-console 1`] = `
"
console.log('hello')
console.log('world')

      ↓ ↓ ↓ ↓ ↓ ↓

const express = require('express');

const cors = require('cors');

const app = express();
const port = process.env.PORT || 3000;
app.use(cors());
app.get('/', (req, res) => {
  res.json('hello');

  console.log('world');
});
app.listen(port, () => {
  console.log('listening on %s', port);
});
"
`;

exports[`api-over-console 6. api-over-console: 6. api-over-console 1`] = `
"
const args = process.argv.slice(2)
console.log({name: args[1]})

      ↓ ↓ ↓ ↓ ↓ ↓

const express = require('express');

const cors = require('cors');

const app = express();
const port = process.env.PORT || 3000;
app.use(cors());
app.get('/', (req, res) => {
  const args = __API_OVER_CONSOLE__obj2Argv(req.query).slice(2);
  res.json({ name: args[1] });
});
app.listen(port, () => {
  console.log('listening on %s', port);
});

function __API_OVER_CONSOLE__obj2Argv(obj) {
  return Object.keys(obj).reduce((prev, key) => [...prev, '--' + key, obj[key]], [null, null]);
}
"
`;

exports[`api-over-console 7. api-over-console: 7. api-over-console 1`] = `
"
console.log()

      ↓ ↓ ↓ ↓ ↓ ↓

console.log();
"
`;
